common/web/gesture-recognizer/src/pathSegmenter.ts

```diff
@@ -0,0 +1,957 @@
+      }
+
+      /**
+       * The total summed squared-distances of the best fitting line from actually-observed values
+       * for their corresponding subsegment; in other words, the "sum of the squared errors" when
+       * utilizing segmented regression.
+       *
+       * Statistically, this is the portion of the dependent variable's variance (un-normalized)
+       * that is unexplained even after dividing the interval into separate subsegments.
+       */
+      get remainingSumSquaredError(): number {
+        return this.pre.sumOfSquaredError + this.post.sumOfSquaredError;
+      }
+
+      /**
+       * Provides the coordinate that belongs to BOTH subsegments, as it ends one and begins
+       * the other.
+       */
+      private get splitPoint() {
+        let splitPoint = this.host.pre.lastSample;
+
+        let indep = splitPoint.t;
+        if(this.independent == 'x') {
+          indep = splitPoint.targetX;
+        } else if(this.independent == 'y') {
+          indep = splitPoint.targetY;
+        }
+
+        let dep = splitPoint.t;
+        if(this.dependent == 'x') {
+          dep = splitPoint.targetX;
+        } else if(this.dependent == 'y') {
+          dep = splitPoint.targetY;
+        }
+
+        return {
+          dep: dep,
+          indep: indep
+        };
+      }
+
+      /**
+       * The total summed squared-distances of the best fitting line from actually-observed values
+       * when **not** utilizing segmented regression.  Double-counts the subsegment's common point, as
+       * it is counted within both subsegments' sum-of-squared error term.
+       */
+      get unsegmentedSumSquaredError(): number {
+        // What was our remaining error for the unsegmented regression?
+        let baseSSE = this.union.sumOfSquaredError;
+
+        // We double-count the split point when segmenting, so we should add an
+        // extra copy of its residual.
+        const splitPoint = this.splitPoint;
+        const splitPointError = this.splitPoint.dep - this.union.predictFromValue(splitPoint.indep);
+
+        // This is the total sum-squared error to be handled by segmenting.
+        return baseSSE + splitPointError * splitPointError;
+      }
+
+      /**
+       * Gets the amount of (unnormalized) variance not explained by a standard regression on
+       * the full interval but succesfully explained by splitting the interval in twain via
+       * segmented regression.
+       */
+      get sumSquaredGainFromSegmentation(): number {
+        return this.unsegmentedSumSquaredError - this.remainingSumSquaredError;
+      }
+
+      /**
+       * A statistical term that signals how successful the segmented regression is.  Always
+       * has values on the interval [0, 1], with 1 being a perfect fit.
+       */
+      get coefficientOfDetermination(): number {
+        if(this.host.union.squaredSum(this.dependent) == 0 || this.host.union.squaredSum(this.independent) == 0) {
```

Comment:
I'm curious -- you test `this.host.union.squaredSum(this.independent)` here but never actually use it later in the function. Why?